zip-0315.rst

```diff
@@ -1,9 +1,629 @@
+
+
+Specification
+=============
+
+Long-term storage of funds
+--------------------------
+
+It is RECOMMENDED that wallets only hold funds as shielded in the long term;
+that is, if a wallet supports receiving transparent funds (or supports
+importing a seed from another wallet that might have done so), then it SHOULD
+either auto-shield or opportunistically shield such funds by default.
+
+Following this recommendation improves both collective privacy and the user's
+individual privacy, by maximizing the size of the note anonymity set over time.
+
+The remainder of this specification assumes a wallet that follows the above
+recommendation, except where explicitly noted.
+
+A wallet MAY allow users to disable auto-shielding.
+
+
+Trusted and untrusted TXOs
+--------------------------
+
+A wallet SHOULD treat received TXOs that are outputs of transactions created
+by the same wallet, as trusted TXOs. Wallets MAY enable users to mark specific
+external transactions as trusted, allowing their received TXOs also to be
+classified as trusted TXOs.
+
+A wallet SHOULD have a policy that is clearly communicated to the user for
+the number of confirmations needed to spend untrusted and trusted TXOs
+respectively. The following confirmation policy is RECOMMENDED:
+
+* 10 confirmations, for untrusted TXOs;
+* 3 confirmations, for trusted TXOs.
+
+Rationale for the given numbers of confirmations
+''''''''''''''''''''''''''''''''''''''''''''''''
+
+The rationale for choosing three confirmations for trusted TXOs is that
+empirically, reorgs are usually less than three blocks.
+
+The consequences of attempting to spend a trusted TXO may be less severe in the
+case of a rollback than the consequences of attempting to spend an untrusted TXO.
+The value received from a trusted TXO should always be recoverable, whereas
+recovering value received from an untrusted TXO may require the user to request
+that funds are re-sent.
+
+
+Categories of TXOs according to spendability
+--------------------------------------------
+
+A TXO is *spendable*, relative to a given block chain and wallet state,
+if and only if all of the following are true in that state:
+
+* the TXO is unspent;
+* the TXO is not committed to be spent in another transaction created
+  by this wallet; and
+* the wallet has the TXO's spending key (for whatever protocol the TXO uses).
+
+A TXO is *confirmed-spendable*, relative to a given block chain and
+wallet state, if and only if all of the following are true in that state:
+
+* the wallet is synchronized; and
+* the TXO is spendable; and
+* either auto-shielding is disabled or the TXO is shielded; and
+* the TXO is trusted and has at least the required confirmations for
+  trusted TXOs, or it is untrusted and has at least the required
+  confirmations for untrusted TXOs.
+
+A TXO is *unconfirmed-spendable*, relative to a given block chain and
+wallet state, if and only if the TXO is spendable but is not
+confirmed-spendable in that state.
```

Comment:
```suggestion
A TXO is *unconfirmed-spendable*, relative to a given block chain and
wallet state, if and only if the TXO is known-spendable but is not
confirmed user-spendable in that state.
```

Reply:
I amended @daira's suggestion to revert `confirmed user-spendable` back to `confirmed-spendable`, as the latter is still the defined term after hir suggestions.